pkg/scheduler/plugins/nodenumaresource/cpu_allocator.go

```diff
@@ -0,0 +1,695 @@
+/*
+Copyright 2022 The Koordinator Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package nodenumaresource
+
+import (
+	"fmt"
+	"sort"
+
+	"github.com/koordinator-sh/koordinator/apis/extension"
+	schedulingconfig "github.com/koordinator-sh/koordinator/apis/scheduling/config"
+)
+
+var getAvailableCPUsFunc = getAvailableCPUs
+
+// getAvailableCPUs returns the available logic CPUs in the node
+func getAvailableCPUs(node *nodeNUMAInfo) (availableCPUs CPUSet, allocated CPUDetails) {
+	if !node.cpuTopology.IsValid() {
+		return
+	}
+	allocated = node.allocatedCPUs.Clone()
+	availableCPUs = node.cpuTopology.CPUDetails.CPUs().Difference(allocated.CPUs())
+	return availableCPUs, allocated
+}
+
+func takeCPUs(
+	topology *CPUTopology,
+	availableCPUs CPUSet,
+	allocatedCPUs CPUDetails,
+	numCPUsNeeded int,
+	cpuBindPolicy extension.CPUBindPolicy,
+	exclusive bool,
+	numaAllocatedStrategy schedulingconfig.NUMAAllocateStrategy,
+) (CPUSet, error) {
+	acc := newCPUAccumulator(topology, availableCPUs, allocatedCPUs, numCPUsNeeded, exclusive, numaAllocatedStrategy)
+	if acc.isSatisfied() {
```

Comment:
This judgment shows that the requested cpu is empty, why is it needed?

Reply:
Just for completeness